crypto/src/obfuscate.rs

```diff
@@ -0,0 +1,99 @@
+use crate::prelude::{ClientId, ConversationId};
+use hex;
+use log::kv::{ToValue, Value};
+use openmls::framing::Sender;
+use openmls::group::QueuedProposal;
+use openmls::prelude::Proposal;
+use sha2::{Digest, Sha256};
+use std::fmt::{Debug, Formatter};
+
+pub(crate) trait Obfuscate {
+    fn obfuscate(&self, f: &mut Formatter<'_>) -> core::fmt::Result;
+}
+
+impl Obfuscate for &ConversationId {
+    fn obfuscate(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        f.write_str(hex::encode(compute_hash(self)).as_str())
+    }
+}
+
+impl Obfuscate for &ClientId {
+    fn obfuscate(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        f.write_str(hex::encode(compute_hash(self)).as_str())
+    }
+}
+
+impl Obfuscate for &Proposal {
+    fn obfuscate(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        f.write_str(match self {
+            Proposal::Add(_) => "Add",
+            Proposal::Update(_) => "Update",
+            Proposal::Remove(_) => "Remove",
+            Proposal::PreSharedKey(_) => "PreSharedKey",
+            Proposal::ReInit(_) => "ReInit",
+            Proposal::ExternalInit(_) => "ExternalInit",
+            Proposal::AppAck(_) => "AppAck",
+            Proposal::GroupContextExtensions(_) => "GroupContextExtensions",
+        })
+    }
+}
+
+impl Obfuscate for &QueuedProposal {
+    fn obfuscate(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        f.write_str(match self.proposal {
```

Comment:
Could reduce code duplication here:

```rust
self.proposal.obfuscate(f)
```

I assume it's intentional that you can't tell from the `Obfuscate` implementation whether this is a `Proposal` instance or a `QueuedProposal`?